neo/core/dataobject.py

```diff
@@ -0,0 +1,328 @@
+# -*- coding: utf-8 -*-
+"""
+This module defines :class:`DataObject`, the abstract base class
+used by all :module:`neo.core` classes that can contain data (i.e. are not container classes).
+It contains basic functionality that is shared among all those data objects.
+
+"""
+import copy
+import warnings
+
+import quantities as pq
+import numpy as np
+from neo.core.baseneo import BaseNeo, _check_annotations
+
+
+class DataObject(BaseNeo, pq.Quantity):
+    '''
+    This is the base class from which all objects containing data inherit
+    It contains common functionality for all those objects and handles array_annotations.
+
+    Common functionality that is not included in BaseNeo includes:
+    - duplicating with new data
+    - rescaling the object
+    - copying the object
+    - returning it as pq.Quantity or np.ndarray
+    - handling of array_annotations
+
+    Array_annotations are a kind of annotations that contain metadata for every data point,
+    i.e. per timestamp (in Spiketrain, Event and Epoch) or signal channel (in AnalogSignal
+    and IrregularlySampledSignal).
+    They may contain the same data as regular annotations, except they are arrays containing these
```

Comment:
This sentence needs clarification, do you mean something like 'Array annotations can contain the same data types as regular annotations, but are always represented as numpy arrays of the same length as the number of data points of the annotated neo object'?

Reply:
Yes, you are right. I'll change this.